platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1126,7 +1126,19 @@ - (id)mgl_jsonExpressionObject {
                 if (firstOp.expressionType == NSConstantValueExpressionType
                     && [firstOp.constantValue isEqualToString:@"format"]) {
                     // Avoid wrapping format options object in literal expression.
-                    return @[@"format", self.arguments[1].mgl_jsonExpressionObject, self.arguments[2].constantValue];
+                    NSMutableArray *expressionObject = [NSMutableArray array];
+                    [expressionObject addObject:@"format"];
+                    
+                    for (NSUInteger index = 1; index < self.arguments.count; index++) {
+                        if (index % 2 == 1) {
+                            [expressionObject addObject:self.arguments[index].mgl_jsonExpressionObject];
```

Comment:
Good catch. Does anything need to be fixed going in the other direction, from JSON to NSExpression?

Reply:
The other way around is ok. The issue is that when is encoded into an `NSExpression` uses different abstract base types; making comparison tests to fail.

I added this test that compares each resulting json object from each parameter:
https://github.com/mapbox/mapbox-gl-native/pull/14064/files#diff-a4216b00e3cfddef98fc22cbc71f3116R1015